### PR TITLE
Add stack compatibility info to Fleet Server docs

### LINKED ISSUE #643: Document stack compatibility for Fleet Server
Confirm that the following is true and it to the prerequisites:

Compatibility: Elasticsearch >= Fleet Server >= Elastic Agent. Bugfix releases might differ. Kibana is assumed to be on the same minor as Elasticsearch.

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #643 and https://github.com/elastic/observability-docs/issues/640

This info was dumped into a meta issue awhile back ago. Need confirmation that the details are correct.

### CODE REVIEW

File: docs/en/ingest-management/fleet/fleet-server.asciidoc [hunk 1]
```diff
@@ -38,18 +38,23 @@ details, refer to <<fleet-server-deployment>>.
...
 
-* {stack} 7.13 or later (our {ess-product}[hosted {ess}] on {ecloud} or
+* {stack} 7.13 or later ({ess-product}[hosted {ess}] on {ecloud}, or
 a self-managed cluster).
+** For version compatibility: {es} >= {fleet-server} >= {agent} (except for
+bugfix releases)
+** {kib} should be on the same minor version as {es}.
 
 * {ece} 2.10 or later
+** Requires additional wildcard domains and certificates (which normally only
```
Comment (ruflin) [inline docs/en/ingest-management/fleet/fleet-server.asciidoc:52]: @simitt @AlexP-Elastic Could you double check this?

Comment (AlexP-Elastic) [reply docs/en/ingest-management/fleet/fleet-server.asciidoc:52]: should this link to the ECE 2.10 docs which should have more details? I'm not sure it's actionable enough as it stands

File: docs/en/ingest-management/fleet/fleet-server.asciidoc [hunk 2]
```diff
@@ -38,18 +38,23 @@ details, refer to <<fleet-server-deployment>>.
...
+cover *.cname, not ..cname). This enables us to provide the URL for
+{fleet-server} of `https://.fleet.`
+** The deployment template must contain a slider for APM & Fleet.
 
-//TODO: Confirm that ECE version shown here is correct ^^.
-
-{fleet-server} is supported on x64 architectures only.
+//QUESTION: Should this say 2.9? 2.10 is not available yet.
```
Comment (AlexP-Elastic) [inline docs/en/ingest-management/fleet/fleet-server.asciidoc:57]: Fleet server won't run on 2.9

Comment (dedemorton) [reply docs/en/ingest-management/fleet/fleet-server.asciidoc:57]: Yah, that's what I figured. So I might need to comment out this section unless ECE 2.10 is shipping soon.

File: docs/en/ingest-management/fleet/fleet-server.asciidoc (hunk 1 above)
Comment (dedemorton) [reply docs/en/ingest-management/fleet/fleet-server.asciidoc:52]: Yes, when the docs are available.

**Review by AlexP-Elastic — APPROVED** [review]
LGTM

**Pull request merged**